src/promptflow/tests/sdk_cli_test/recording_utilities/record_storage.py

```diff
@@ -161,7 +158,107 @@ def _recursive_create_hashable_args(self, item):
         else:
             return item
 
-    def set_record(self, input_dict: Dict, output) -> None:
+    def _parse_output_generator(self, output):
+        """
+        Special handling for generator type. Since pickle will not work for generator.
+        Returns the real list for reocrding, and create a generator for original output.
+        Parse output has a simplified hypothesis: output is simple dict, list or generator,
+        because a full schema of output is too heavy to handle.
+        Example: {"answer": <generator>, "a": "b"}, [<generator>, a, b], <generator>
+        """
+        output_type = ""
+        output_value = None
+        output_generator = None
+        if isinstance(output, dict):
+            output_value = {}
+            output_generator = {}
+            for item in output.items():
+                k, v = item
+                if type(v).__name__ == "generator":
+                    vlist = list(v)
+
+                    def vgenerator():
+                        for vitem in vlist:
+                            yield vitem
+
+                    output_value[k] = vlist
+                    output_generator[k] = vgenerator()
+                    output_type = "dict[generator]"
+                else:
+                    output_value[k] = v
+        elif isinstance(output, list):
```

Comment:
I think we don't have scenario for list, we might remove this branch

Reply:
OKay